code/modules/assembly/infrared.dm

```diff
@@ -246,7 +246,9 @@
 	hit()
 
 /obj/effect/beam/i_beam/Crossed(atom/movable/AM as mob|obj)
-	if(istype(AM, /obj/effect/beam) || !AM.density)
+	if(!isobj(AM) && !isliving(AM))
+		return
+	if(istype(AM, /obj/effect/beam))

```

Comment:
@Krausus Will get on it

@Crazylemon64 Well, blood and gibs are physical things that should, therefore, trip a physical sensor.